In [1]:
from google.colab import files
uploaded = files.upload()

Saving Indian_Kids_Screen_Time.csv to Indian_Kids_Screen_Time.csv


In [2]:
import pandas as pd

In [4]:
import pandas as pd
import numpy as np

df = pd.read_csv("Indian_Kids_Screen_Time.csv")
print(df.shape)
df.head()

(9712, 8)


Age  Gender  Avg_Daily_Screen_Time_hr Primary_Device  \
0   14    Male                      3.99     Smartphone   
1   11  Female                      4.61         Laptop   
2   18  Female                      3.73             TV   
3   15  Female                      1.21         Laptop   
4   12  Female                      5.89     Smartphone   

   Exceeded_Recommended_Limit  Educational_to_Recreational_Ratio  \
0                        True                               0.42   
1                        True                               0.30   
2                        True                               0.32   
3                       False                               0.39   
4                        True                               0.49   

           Health_Impacts Urban_or_Rural  
0  Poor Sleep, Eye Strain          Urban  
1              Poor Sleep          Urban  
2              Poor Sleep          Urban  
3                     NaN          Urban  
4     Poor Sleep, Anxiety          Urban

In [5]:
print(df.isnull().sum())

Age                                     0
Gender                                  0
Avg_Daily_Screen_Time_hr                0
Primary_Device                          0
Exceeded_Recommended_Limit              0
Educational_to_Recreational_Ratio       0
Health_Impacts                       3218
Urban_or_Rural                          0
dtype: int64


In [7]:
duplicate_count=df.duplicated().sum()
print(duplicate_count)

df = df.drop_duplicates()
print(df.shape)

44
(9668, 8)


In [14]:
df.describe()


Age  Avg_Daily_Screen_Time_hr  \
count  9668.000000               9668.000000   
mean     12.992760                  4.367780   
std       3.158931                  1.703002   
min       8.000000                  0.000000   
25%      10.000000                  3.427500   
50%      13.000000                  4.440000   
75%      16.000000                  5.380000   
max      18.000000                 13.890000   

       Educational_to_Recreational_Ratio  
count                        9668.000000  
mean                            0.427028  
std                             0.073185  
min                             0.300000  
25%                             0.370000  
50%                             0.430000  
75%                             0.480000  
max                             0.600000

In [8]:
import matplotlib.pyplot as plt
import seaborn as sns

In [9]:
df.corr(numeric_only=True)

Age  Avg_Daily_Screen_Time_hr  \
Age                                1.000000                  0.108873   
Avg_Daily_Screen_Time_hr           0.108873                  1.000000   
Exceeded_Recommended_Limit         0.150615                  0.657812   
Educational_to_Recreational_Ratio -0.488017                 -0.081445   

                                   Exceeded_Recommended_Limit  \
Age                                                  0.150615   
Avg_Daily_Screen_Time_hr                             0.657812   
Exceeded_Recommended_Limit                           1.000000   
Educational_to_Recreational_Ratio                   -0.120873   

                                   Educational_to_Recreational_Ratio  
Age                                                        -0.488017  
Avg_Daily_Screen_Time_hr                                   -0.081445  
Exceeded_Recommended_Limit                                 -0.120873  
Educational_to_Recreational_Ratio                           1.000000

In [12]:
Q1 = df['Avg_Daily_Screen_Time_hr'].quantile(0.25)
Q3 = df['Avg_Daily_Screen_Time_hr'].quantile(0.75)
IQR = Q3 - Q1
print("IQR:", IQR)

IQR: 1.9524999999999997


Dataset had 9712 rows and 8 columns, after removing duplicates, 9668 rows remain.
Health_Impacts column has 3218 missing values.
44 duplicate rows were removed.
Calculates the interquartile range(IQR) of daily screen time to help detect outliers.